### PR TITLE
Add GitHub status formatter

### PR DESCRIPTION
**Pull request by mknapik**
Solves #121.

The formatter creates one commit status per runner with information about issue counts.
Issues in status description are ordered by severity.

By default all Pronto Messages with level `:warning` or higher are considered a failed state for GitHub status (red status).
`GithubStatusFormatter` allows to override this behaviour by accepting `:level_mapping` option in the initializer (which is not supported when running the formatter directly with command line, e.g. `pronto run -f github_pr_status`).
There is no README info yet but I'll provide it once we agree on the interface and the way this formatter should work.

The result of running Pronto with the new formatter:
GithubStatusFormatter

@mmozuras Please let me know whether the PR adheres to the project conventions.
I got some RuboCop `line to long warnings` in specs, should I fix those?
Also I'm going to squash refactoring commits before merging.

### CODE REVIEW

File: lib/pronto/formatter/github_status_formatter.rb [hunk 1]
```diff
@@ -0,0 +1,85 @@
+require_relative 'github_status_formatter/sentence'
+
+module Pronto
+  module Formatter
+    class GithubStatusFormatter
+      def initialize(opts = {})
+        @level_mapping ||= opts.fetch(:level_mapping, {})
```
Comment (mmozuras) [inline lib/pronto/formatter/github_status_formatter.rb]: Not sure if it's worth having a way to override it. Let's leave it off for now 😄

File: lib/pronto/formatter/github_status_formatter/sentence.rb [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+        def to_s
+          case words.size
+          when 0
+            ''
+          when 1
+            words[0].to_s.dup
+          when 2
+            "#{words[0]}#{WORD_CONNECTORS[:two_words_connector]}#{words[1]}"
+          else
+            to_oxford_comma_sentence
```
Comment (mmozuras) [inline lib/pronto/formatter/github_status_formatter/sentence.rb]: Why have you chosen oxford comma?

**Comment by mmozuras** [comment]
@mknapik amazing, thank you, honestly 👏 😄 

Just address a couple of comments, add info to README and CHANGELOG and it's gonna get merged! 😄

File: lib/pronto/formatter/github_status_formatter.rb (hunk 1 above)
Comment (mknapik) [inline lib/pronto/formatter/github_status_formatter.rb]: The reason for implementing this was that Reek was returning only `warning`s even though stuff like `Doesn't depend on instance state (maybe move it to another class?) (UtilityFunction)` sound pretty subjective to me. Now when I think about it, it'd probably be better to address the issue at mmozuras/pronto-reek than here.

I'll get rid of `opts` parameter for now.

File: lib/pronto/formatter/github_status_formatter/sentence.rb (hunk 2 above)
Comment (mknapik) [inline lib/pronto/formatter/github_status_formatter/sentence.rb]: I shamelessly ripped off and simplified the implementation from [ActiveSupport](http://apidock.com/rails/ActiveSupport/CoreExtensions/Array/Conversions/to_sentence).

**Comment by mknapik** [comment]
@mmozuras Apart from adhering to your remarks I've also did some refactoring.
Additionally I renamed formatter cli name from `github_pr_status` to `github_status`.
Let me know what do you think.
I haven't squashed everything yet so you can review the changes easily.

File: README.md [hunk 3]
```diff
@@ -103,13 +103,21 @@ or, if you want comments to appear on pull request diff, instead of commit:
 $ GITHUB_ACCESS_TOKEN=token PULL_REQUEST_ID=id pronto run -f github_pr -c origin/master
 ```
 
+Use `GithubStatusFormatter` to submit [commit status](https://github.com/blog/1227-commit-status-api):
+
+```sh
+$ GITHUB_ACCESS_TOKEN=token pronto run -f github_status -c origin/master
```
Comment (mmozuras) [inline README.md]: I think that it's worth showing that multiple formatters can be used here.

**Comment by mmozuras** [comment]
@mknapik looks good, just change that one last thing 😄

**Comment by aergonaut** [comment]
@mknapik @mmozuras would it be too much to ask to have the status contexts formatted more like how other services format them? i.e. instead of "Pronto::Rubocop", something like "pronto/rubocop" or "lint/pronto/rubocop" to be more similar to how Travis CI uses "continuous-integration/travis-ci/pr"

**Comment by mknapik** [comment]
@aergonaut Actually I had no idea what was the convention.
The [docs](https://developer.github.com/v3/repos/statuses/) use `continuous-integration/jenkins` and `security/brakeman` as examples.
I guess I can change it to `pronto/$runner`.

File: README.md (hunk 3 above)
Comment (mknapik) [inline README.md]: Ok, I'll add one more example.
Although I believe using multiple formatters is not documented anywhere else.

**Comment by mknapik** [comment]
@mmozuras @aergonaut Regarding context names:
Do you think a simple function like `runner.class.downcase.gsub('::', '/')` is enough or should I implement something like [`ActiveSupport::Inflector#underscore`](http://apidock.com/rails/v4.2.1/ActiveSupport/Inflector/underscore)?

The difference:

Pronto Runner | downcase | underscore
--- | --- | ---
`Pronto::Brakeman` | `pronto/brakeman` | `pronto/brakeman`
`Pronto::RailsBestPractices` | `pronto/railsbestpractices` | `pronto/rails_best_practices`

**Comment by aergonaut** [comment]
@mknapik looks good to me 😄

**Comment by mmozuras** [comment]
@mknapik looks good, amazing contribution 👍 🙇

**Comment by mknapik** [comment]
@mmozuras Thanks!
The gem is well written and contributing was a pleasure!

**Comment by kevinjalbert** [comment]
This is awesome. Was trying to make it work using version 0.6.0, then realized that this isn't released yet 😉 .

I look forward to using it in the next version bump.

**Pull request merged**